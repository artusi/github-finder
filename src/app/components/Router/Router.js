import React from "react";
import PropTypes from "prop-types";
import { BrowserRouter, Switch, Route } from "react-router-dom";

function Router({ routes, children }) {
  return (
    <BrowserRouter>
      <Switch>
        {routes.map(({ path, type, ...routeProps }) => (
          <Route key={path} path={path} type={type} {...routeProps} />
        ))}
        {children}
      </Switch>
    </BrowserRouter>
  );
}

Router.propTypes = {
  routes: PropTypes.arrayOf(
    PropTypes.shape({
      path: PropTypes.string.isRequired
    }).isRequired
  ).isRequired,
  children: PropTypes.any
};
Router.defaultProps = {
  children: null
};

export default Router;
