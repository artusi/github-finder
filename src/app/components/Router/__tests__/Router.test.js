import React from "react";
import Router from "../Router";

describe("<Router />", () => {
  const routes = [
    {
      path: "/",
      exact: true,
      component: () => <span>Test</span>
    },
    {
      path: "/test-route",
      component: () => <span>Test2</span>
    },
    {
      path: "/test-another-route",
      component: () => <span>Test2</span>
    }
  ];

  it("renders > empty", () => {
    const wrapper = shallow(<Router routes={[]} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders > list", () => {
    const wrapper = shallow(<Router routes={routes} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders > list with children", () => {
    const wrapper = shallow(
      <Router routes={routes}>
        <span />
      </Router>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
