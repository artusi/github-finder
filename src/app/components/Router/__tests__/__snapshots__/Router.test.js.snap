// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Router /> renders > empty 1`] = `
<BrowserRouter>
  <Switch />
</BrowserRouter>
`;

exports[`<Router /> renders > list 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      key="/"
      path="/"
    />
    <Route
      component={[Function]}
      key="/test-route"
      path="/test-route"
    />
    <Route
      component={[Function]}
      key="/test-another-route"
      path="/test-another-route"
    />
  </Switch>
</BrowserRouter>
`;

exports[`<Router /> renders > list with children 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      key="/"
      path="/"
    />
    <Route
      component={[Function]}
      key="/test-route"
      path="/test-route"
    />
    <Route
      component={[Function]}
      key="/test-another-route"
      path="/test-another-route"
    />
    <span />
  </Switch>
</BrowserRouter>
`;
